### PR TITLE
Creating admin panel

### PR DESCRIPTION
**Pull request by konrad04-1989**

### CODE REVIEW

File: spec/factories/admins.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :admin do
+    email { 'admin@admin.com' }
```
Comment (ciejj) [inline spec/factories/admins.rb]: I suggest using sequence here - just like for the user

File: spec/system/admin_system_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  # rubocop:disable RSpec/ExampleLength
+  it 'enables Admin to sign in' do
+    admin = create(:admin)
+    visit root_path
+    click_link 'Admin Log in'
+    fill_in 'Email', with: admin.email
+    fill_in 'Password', with: admin.password
+    click_button 'Log in'
+    expect(page).to have_text('Signed in successfully.')
```
Comment (ciejj) [inline spec/system/admin_system_spec.rb:18]: It would be good to check not only the notification, but also the landing page

File: db/seeds.rb [hunk 3]
```diff
@@ -17,6 +10,14 @@
 
 puts "Employees table seeding done"
 
+puts "Admins table seeding..."
+    admin = Admin.new
+    admin.email = "admin@rera.com"
+    admin.password = 'zaq123'
+    admin.password_confirmation = 'zaq123'
+    admin.save!
```
Comment (ciejj) [inline db/seeds.rb]: Consider using  `create!` method instead.
Also - do you really need to provide the `password_confirmation` during the object creation? 

```suggestion
    admin.create!(email: 'admin@rera.com', password: 'zaq123')
```

File: config/routes.rb [hunk 4]
```diff
@@ -1,9 +1,15 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  devise_for :admins, path: 'admins'
+  devise_for :employees, path: 'employees'
```
Comment (ciejj) [inline config/routes.rb]: I think, that specifying path in both is not really necessary

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -28,6 +28,16 @@
             <%= link_to 'Log out', destroy_employee_session_path, method: :delete, :onclick => "return confirm('Are you sure you want to Log out?')", class: 'btn btn-outline-danger' %>
           </li>
 
+        <% elsif admin_signed_in? %>
```
Comment (ciejj) [inline app/views/shared/_navbar.html.erb]: This should go to another (admin) navbar - in my understanding on the main page we are just checking if user is logged in

File: app/controllers/concerns/accessible.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Accessible
```
Comment (ciejj) [inline app/controllers/concerns/accessible.rb]: I do not think we need this module at all, also it rather should go to the app/concerns folder, not controllers :)

File: config/routes.rb [hunk 7]
```diff
@@ -1,9 +1,15 @@
...
+
+  authenticated :admin do
```
Comment (ciejj) [inline config/routes.rb]: it should be `namespace` - you should be able to access that path, and login afterwards

**Review by ciejj — CHANGES_REQUESTED** [review]
You did a great job, but I am afraid, that some of it was unnecessary :/ 
In my understanding you have implemented an option to login as an employee or admin, but at this point you should not really care about it - there rather should be a seprate /admin path that requires admin credential.

We would need Oskar's confirmation on this ;)

**Comment by konrad04-1989** [comment]
@ciejj  please take a look if you will have a moment. I have made the changes as per the comments above. It seems to me that it is now working as intended ;)

**Review by ciejj — APPROVED** [review]
Looks good - I would only consider using separate layout for admin panel

**Review by oskarlakner — APPROVED** [review]
well done Konrad!

**Comment by konrad04-1989** [comment]
Thanks Guys for the approve!
@ciejj   I'll correct separate layout for admin in the next sprint.

**Pull request merged**